src/qilib/utils/serialization.py

```diff
@@ -20,18 +20,68 @@
 import base64
 from functools import partial
 from json import JSONDecoder, JSONEncoder
-from typing import Any, Callable, Dict, Tuple, Optional
+from typing import Any, Callable, Dict, Tuple, Optional, Union
 from dataclasses_json.api import DataClassJsonMixin
-
 import numpy as np
 
-from qilib.data_set.mongo_data_set_io import MongoDataSetIO
+from qilib.utils.type_aliases import EncodedNumpyArray, numpy_ndarray_type, NumpyKeys
+
 
 # A callable type for transforming a given argument with a type to another type
 TransformFunctionResult = Any
 TransformFunction = Callable[[Any], TransformFunctionResult]
 
 
+class NumpyArrayEncDec:
+    """ Helper class for the numpy array decoder and encoder """
+    @staticmethod
+    def encode_numpy_array(array: numpy_ndarray_type, encode_to_bytes: bool = False) -> EncodedNumpyArray:
```

Comment:
Using a boolean parameter to determine behaviour is a code smell ([example article](https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/)). Does it make sense to have separate `encode_numpy_array_to_bytes` and `encode_numpy_array_to_base64` methods? That way the behaviour is much more descriptive, and you can still reuse part of the logic using a private method.

Reply:
This was a change from another open merge request (https://github.com/QuTech-Delft/qilib/pull/111) I included blindly. I agree we should make it more explicit. I kept the old name encode_numpy_array but added encode_numpy_array_to_bytes